ofrak_core/ofrak/gui/server.py

```diff
@@ -739,6 +742,61 @@ async def search_for_vaddr(self, request: Request) -> Response:
         except NotFoundError:
             return json_response([])
 
+    @exceptions_to_http(SerializedError)
+    async def search_for_string(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        string_query_string = body["search_query"]
+        regex = body["regex"]
+        case_ignore = body["case_ignore"]
+        if not isinstance(string_query_string, str):
+            raise ValueError("Invalid search query.")
+        string_query: Union[bytes, re.Pattern[bytes]] = string_query_string.encode()
+        if case_ignore:
+            string_query = re.compile(string_query, re.IGNORECASE)
+        elif regex:
+            string_query = re.compile(string_query)
+
+        offsets = await resource.search_data(string_query)
+        if string_query == "":
+            return json_response(None)
+        found_resources = []
+        if len(offsets) > 0:
+            # found_resources[resource.get_id().hex()] = offsets
+            found_resources.append(resource.get_id().hex())
+        for child in await resource.get_descendants():
+            offsets = await child.search_data(string_query)
+            if len(offsets) > 0:
+                # found_resources[child.get_id().hex()] = offsets
+                found_resources.append(child.get_id().hex())
+                for ancestor in await child.get_ancestors():
+                    found_resources.append(ancestor.get_id().hex())
+
+        return json_response(found_resources)
+
+    @exceptions_to_http(SerializedError)
+    async def search_for_bytes(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        bytes_query_request = body["search_query"]
+        if bytes_query_request == "":
+            return json_response(None)
+        bytes_query = bytes.fromhex("".join(bytes_query_request.split(" ")))
```

Comment:
Why split and join instead of just removing all non-hex characters?

```suggestion
        bytes_query = bytes.fromhex(re.sub(r"[^0-9a-fA-F]+", "", bytes_query_request))
```

Reply:
It looks nicer, easier to read for long strings of bytes, much of our hex input follows this format.